diff --git a/src/python/grpcio/grpc_core_dependencies.py b/src/python/grpcio/grpc_core_dependencies.py
index 21efd68287..98a850101d 100644
--- a/src/python/grpcio/grpc_core_dependencies.py
+++ b/src/python/grpcio/grpc_core_dependencies.py
@@ -703,6 +703,7 @@ CORE_SOURCE_FILES = [
     'third_party/cares/cares/ares_timeout.c',
     'third_party/cares/cares/ares_version.c',
     'third_party/cares/cares/ares_writev.c',
+    'third_party/cares/cares/ares_android.c',
     'third_party/cares/cares/bitncmp.c',
     'third_party/cares/cares/inet_net_pton.c',
     'third_party/cares/cares/inet_ntop.c',
diff --git a/third_party/cares/config_linux/ares_config.h b/third_party/cares/config_linux/ares_config.h
index 3634e9d061..cdcefd817c 100644
--- a/third_party/cares/config_linux/ares_config.h
+++ b/third_party/cares/config_linux/ares_config.h
@@ -128,7 +128,7 @@
 #define HAVE_GETNAMEINFO

 /* Define to 1 if you have the getservbyport_r function. */
-#define HAVE_GETSERVBYPORT_R
+//#define HAVE_GETSERVBYPORT_R

 /* Define to 1 if you have the `gettimeofday' function. */
 #define HAVE_GETTIMEOFDAY
@@ -444,7 +444,7 @@
   #undef FD_ISSET
   /* 'FD_ZERO' doesn't use __fdelt_chk, no need to redefine. */

-  #ifdef __FDS_BITS
+  #ifndef __FDS_BITS
     #define GRPC_CARES_FDS_BITS(set) __FDS_BITS(set)
   #else
     #define GRPC_CARES_FDS_BITS(set) ((set)->fds_bits)
